mne_bids/mne_bids.py

```diff
@@ -570,6 +570,16 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
         raise ValueError('raw_file must be an instance of str or BaseRaw, '
                          'got %s' % type(raw_file))
 
+    if 'meg' in raw:
+        kind = 'meg'
+    elif 'eeg' in raw:
+        kind = 'eeg'
+    elif 'ecog' in raw:
+        kind = 'ecog'
+    else:
+        raise ValueError('Neither MEG/EEG/iEEG channels found in data.'
+                         'Please set "kind" explicitly.')
+
```

Comment:
this if/else part needs to be skipped if kind was set explicitly ... that is, only use this code, if the argument kind =='auto'